File: src/main/java/net/artux/pdanetwork/service/communication/ConversationService.java
package net.artux.pdanetwork.service.communication;

import net.artux.pdanetwork.models.communication.ConversationCreateDTO;
import net.artux.pdanetwork.models.communication.ConversationDTO;
import net.artux.pdanetwork.models.page.QueryPage;
import net.artux.pdanetwork.models.page.ResponsePage;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.UUID;

public interface ConversationService {

    ConversationDTO createConversation(ConversationCreateDTO createDTO);

    ConversationDTO editConversation(UUID id, ConversationCreateDTO createDTO);

    ConversationDTO getConversation(UUID id);
    List<ConversationDTO> getConversationWithUser(UUID userId);

    Page<ConversationDTO> getConversations(Pageable queryPage);

    ResponsePage<ConversationDTO> getConversations(QueryPage queryPage);

    boolean deleteConversation(UUID id);

    ConversationDTO getConversationEntity(UUID id);
}

File: pdanet/src/main/java/net/artux/pdanetwork/service/communication/ConversationService.kt
package net.artux.pdanetwork.service.communication

import net.artux.pdanetwork.dto.page.QueryPage
import net.artux.pdanetwork.dto.page.ResponsePage
import net.artux.pdanetwork.models.communication.ConversationCreateDTO
import net.artux.pdanetwork.models.communication.ConversationDTO
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import java.util.UUID

interface ConversationService {
    fun createConversation(createDTO: ConversationCreateDTO): ConversationDTO
    fun editConversation(id: UUID, createDTO: ConversationCreateDTO): ConversationDTO
    fun getConversation(id: UUID): ConversationDTO
    fun getConversationWithUser(userId: UUID): List<ConversationDTO>
    fun getConversations(queryPage: Pageable): Page<ConversationDTO>
    fun getConversations(queryPage: QueryPage): ResponsePage<ConversationDTO>
    fun deleteConversation(id: UUID): Boolean
    fun getConversationEntity(id: UUID): ConversationDTO
    fun deleteAllConversationsWithUser(id: UUID): Boolean
}
